src/metrics/include/metrics/operation.hpp

```diff
@@ -123,13 +147,24 @@ class OperationImpl final : private boost::noncopyable {
         return _events;
     }
 
-    void reportAt(time_point finished, OperationEvent<ClockSource>&& event) {
+    void reportAt(time_point started, time_point finished, OperationEvent<ClockSource>&& event) {
+        if (_threshold) {
```

Comment:
This is the only extra `if` statement that happens at runtime. I thought it was better to take this hit than to template all the way up to the metrics reporter.